sklearn/preprocessing/_encoders.py

```diff
@@ -213,14 +255,56 @@ class OneHotEncoder(_BaseEncoder):
     dtype : number type, default=np.float
         Desired dtype of output.
 
-    handle_unknown : {'error', 'ignore'}, default='error'
+    handle_unknown : {'error', 'ignore', 'auto'}, default='error'
         Whether to raise an error or ignore if an unknown categorical feature
         is present during transform (default is to raise). When this parameter
         is set to 'ignore' and an unknown category is encountered during
         transform, the resulting one-hot encoded columns for this feature
         will be all zeros. In the inverse transform, an unknown category
         will be denoted as None.
 
+        When this parameter is set to 'auto' and an unknown category is
+        encountered in transform:
+
+            1. If infrequent category support was not configured or there were
+            no infrequent category during training, the resulting one-hot
+            encoded columns for this feature will be all zeros. In the inverse
+            transform, an unknown category will be denoted as `None`.
+
+            2. If there is an infrequent category during training, the unknown
+            category will be considered infrequent. In the inverse transform,
+            'infrequent' will be used to represent the infrequent category. If
+            'infrequent' is already a category, 'infrequent_sklearn' will be
```

Comment:
This information about how name collisions are handled probably should be somewhere else, right? It doesn't really have to do with the handle_unknown parameter.